core/src/test/java/de/jplag/merging/MergingTest.java

```diff
@@ -0,0 +1,137 @@
+package de.jplag.merging;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import de.jplag.GreedyStringTiling;
+import de.jplag.JPlagComparison;
+import de.jplag.JPlagResult;
+import de.jplag.Match;
+import de.jplag.SubmissionSet;
+import de.jplag.SubmissionSetBuilder;
+import de.jplag.TestBase;
+import de.jplag.exceptions.ExitException;
+import de.jplag.options.JPlagOptions;
+import de.jplag.strategy.ComparisonStrategy;
+import de.jplag.strategy.ParallelComparisonStrategy;
+
+class MergingTest extends TestBase {
+    private JPlagOptions options;
+    private JPlagResult result;
+    private List<Match> matches;
+    private List<JPlagComparison> comparisonsBefore;
+    private List<JPlagComparison> comparisonsAfter;
+
+    MergingTest() throws ExitException {
+        options = getDefaultOptions("merging").withMergingParameters(new MergingParameters(8, 2));
+
+        GreedyStringTiling coreAlgorithm = new GreedyStringTiling(options);
+        ComparisonStrategy comparisonStrategy = new ParallelComparisonStrategy(options, coreAlgorithm);
+
+        SubmissionSetBuilder builder = new SubmissionSetBuilder(options);
+        SubmissionSet submissionSet = builder.buildSubmissionSet();
+
+        result = comparisonStrategy.compareSubmissions(submissionSet);
+        comparisonsBefore = result.getAllComparisons();
+
+        result = new MatchMerging(result, options).run();
+        comparisonsAfter = result.getAllComparisons();
```

Comment:
Test state should not be shared between test cases, use @BeforeEach and a dedicated method.
Also, we could use a constant for the merging parameter record to emphasize them.

Reply:
Could you elaborate a bit?
I am not that firm with JUnit

Reply:
The confusing thing is, that they changed the names between JUnit 4 and 5, but this guide should help: https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall#beforeeach-and-beforeall

Reply:
The test subject should never be shared between test cases, as the introduction of a bug to the match merging might change its behavior. At the very least lines 44 and 45 should be created before each test.